### PR TITLE
fix: Bump copyright date for each source

### LINKED ISSUE #55: Add copyright declaration for all source
Some files have no copyright message in the header. We should add for them.

### PR DESCRIPTION
**Pull request by splasky**
Add copyright license if it wasn't included in header file.

Closes #55

### CODE REVIEW

**Review by howjmay — APPROVED** [review]

**Comment by jserv** [comment]
You don't have to mention the bump of copyright date within commit messages.

**Comment by jserv** [comment]
You have to express the motivation and considerations for this pull request.

File: utils/crypto_utils.c [hunk 1]
```diff
@@ -155,8 +163,8 @@ int aes_decrypt(const unsigned char *ciphertext, int ciphertext_len, const unsig
   return -1;
 }
 
-int encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *ciphertext, int ciphertext_len,
-            uint8_t iv[AES_BLOCK_SIZE], uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]) {
+int encrypt(const char *plaintext, int plaintext_len, char *ciphertext, int ciphertext_len, uint8_t iv[AES_BLOCK_SIZE],
+            uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]) {
```
Comment (howjmay) [inline utils/crypto_utils.c]: How about add `#define AES_CBC_KEY_SIZE AES_BLOCK_SIZE * 2`?

File: utils/crypto_utils.h [hunk 2]
```diff
@@ -21,14 +21,14 @@ extern "C" {
 
 int get_device_id(const char *device_id);
 int get_aes_key(const uint8_t *key);
-int encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *ciphertext, int ciphertext_len, uint8_t iv[16],
+int encrypt(const char *plaintext, int plaintext_len, char *ciphertext, int ciphertext_len, uint8_t iv[16],
```
Comment (howjmay) [inline utils/crypto_utils.h]: use macro for the legnth of array `iv`

File: utils/crypto_utils.h [hunk 3]
```diff
@@ -21,14 +21,14 @@ extern "C" {
...
             uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]);
-int decrypt(unsigned char *ciphertext, int ciphertext_len, unsigned char *plaintext, int plaintext_len, uint8_t iv[16],
+int decrypt(const char *ciphertext, int ciphertext_len, char *plaintext, int plaintext_len, uint8_t iv[16],
```
Comment (howjmay) [inline utils/crypto_utils.h]: ditto

**Review by howjmay — CHANGES_REQUESTED** [review]

**Review by howjmay — CHANGES_REQUESTED** [review]
Fix typo

* Fix const qualifier warning and pointer signedness warning from
test_crypto_utils. Inside test_crypto_utils.c we defined text test data
as constant char* type, we should unify them.

* Modify iv array constant as macro to unify array length.

**Comment by jserv** [comment]
Hint: You can use the service provided by [LanguageTool](https://languagetool.org/) for checking grammar, style and spell.

**Review by howjmay — CHANGES_REQUESTED** [review]
Could you add corresponding description of the functions in comment

**Comment by jserv** [comment]
Concentrate on single functional change.

**Comment by jserv** [comment]
You should have a pull request which fixes compilation warnings. Another is dedicated to documentation enhancements.

File: utils/crypto_utils.h [hunk 4]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+/**
+ * @brief Gets device IMSI or physical ID
+ *
+ * @param[out] device_id Device id output
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
 int get_device_id(const char *device_id);
```
Comment (howjmay) [inline utils/crypto_utils.h]: We will have `get_device_id()` in HAL. Why do we need an extra more similar function here?

File: utils/crypto_utils.h [hunk 5]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+/**
+ * @brief Gets aes key from hashchain
+ *
+ * @param[out] key Device key output
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
 int get_aes_key(const uint8_t *key);
```
Comment (howjmay) [inline utils/crypto_utils.h]: Return value should be in `retcode_t`

File: utils/crypto_utils.h [hunk 6]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+ * @param[out] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
+ * @param[in] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[in] key[AES_CBC_KEY_SIZE] Decryption key
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
+int decrypt(const char *ciphertext, int ciphertext_len, char *plaintext, int plaintext_len, uint8_t iv[AES_BLOCK_SIZE],
```
Comment (howjmay) [inline utils/crypto_utils.h]: ditto

File: utils/crypto_utils.h [hunk 7]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+ * @param[in] keybits Length of decryption key
+ * @param[in] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[out] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
+int aes_decrypt(const char *ciphertext, int ciphertext_len, const unsigned char *key, unsigned int keybits,
```
Comment (howjmay) [inline utils/crypto_utils.h]: ditto

File: utils/crypto_utils.h [hunk 8]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+ * @param[in] keybits Length of key
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[out] ciphertext Ciphertext
+ * @param[in] ciphertext_len Ciphertext length
+ *
+ * @return
+ * - Ciphertext new length on success
+ * - -1 on error
+ */
+int aes_encrypt(const char *plaintext, int plaintext_len, const unsigned char *key, unsigned int keybits,
```
Comment (howjmay) [inline utils/crypto_utils.h]: ditto

File: utils/crypto_utils.h [hunk 9]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+/**
+ * @brief Implementation of aes encrypt
+ *
+ * @param[in] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
+ * @param[in] key Encryption key
+ * @param[in] keybits Length of key
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[out] ciphertext Ciphertext
+ * @param[in] ciphertext_len Ciphertext length
```
Comment (howjmay) [inline utils/crypto_utils.h]: Don't put a input parameter after a output parameter. The order will mislead people

File: utils/crypto_utils.h [hunk 10]
```diff
@@ -16,19 +16,102 @@ extern "C" {
...
+/**
+ * @brief Implementation of aes decrypt
+ *
+ * @param[in] ciphertext Ciphertext
+ * @param[in] ciphertext_len Ciphertext length
+ * @param[in] key Decryption key
+ * @param[in] keybits Length of decryption key
+ * @param[in] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[out] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
```
Comment (howjmay) [inline utils/crypto_utils.h]: ditto

**Review by howjmay — CHANGES_REQUESTED** [review]

**Comment by splasky** [comment]
This PR mixed document comment and copyright. I will drop the document comment commit.
The document comment issue has already created at #8.

File: utils/crypto_utils.h (hunk 4 above)
Comment (splasky) [reply utils/crypto_utils.h]: Will be removed at #46

File: utils/crypto_utils.h (hunk 8 above)
Comment (splasky) [reply utils/crypto_utils.h]: Need to return new length of  ciphertext.

File: utils/crypto_utils.h (hunk 9 above)
Comment (splasky) [reply utils/crypto_utils.h]: New issue #54 for this.

File: utils/crypto_utils.h (hunk 10 above)
Comment (splasky) [reply utils/crypto_utils.h]: New issue #54 for this.

**Review by howjmay — APPROVED** [review]

**Pull request merged**